src/shared/config.ts

```diff
@@ -24,60 +24,145 @@
  - Name Surname <name.surname@gatesfoundation.com>
 
  - Paweł Marzec <pawel.marzec@modusbox.com>
- - Abhimanyu Kapur <abhi.kapur09@gmail.com>
+ - Raman Mangla <ramanmangla@google.com>
+ - Kenneth Zeng <kkzeng@google.com>
  --------------
  ******/
 
-import rc from 'rc'
-import parse from 'parse-strings-in-object'
-import Config from '../../config/default.json'
-import Package from '../../package.json'
-export interface ServiceConfig {
-  // package.json
-  PACKAGE: object;
+import Convict from 'convict'
+import DbConfig from '~/../config/knexfile'
+import PACKAGE from '../../package.json'
 
-  // ../server.ts
-  PORT: number;
-  HOST: string;
+interface DbConnection {
+  host: string;
+  port: number;
+  user: string;
+  password: string;
+  database: string;
+  timezone: string;
+}
 
-  PARTICIPANT_ID: string;
+interface DbPool {
+  min: number;
+  max: number;
+  acquireTimeoutMillis: number;
+  createTimeoutMillis: number;
+  destroyTimeoutMillis: number;
+  idleTimeoutMillis: number;
+  reapIntervalMillis: number;
+  createRetryIntervalMillis: number;
+}
+
+interface DatabaseConfig {
+  client: string;
+  version?: string;
+  useNullAsDefault?: boolean;
+  connection: DbConnection | string;
+  pool?: DbPool;
 
-  // ../lib/config.
-  DB_ENVIRONMENT: string;
-  DATABASE?: {
-    ACQUIRE_TIMEOUT_MILLIS: number;
-    CREATE_RETRY_INTERVAL_MILLIS: number;
-    CREATE_TIMEOUT_MILLIS: number;
-    DEBUG: boolean;
-    DESTROY_TIMEOUT_MILLIS: number;
-    DIALECT: string;
-    HOST: string;
-    IDLE_TIMEOUT_MILLIS: number;
-    PASSWORD: string;
-    POOL_MAX_SIZE: number;
-    POOL_MIN_SIZE: number;
-    PORT: number;
-    REAP_INTERVAL_MILLIS: number;
-    SCHEMA: string;
-    USER: string;
+  migrations: {
+    directory: string;
+    tableName: string;
+    stub?: string;
+    loadExtensions: string[];
   };
 
-  MIGRATIONS?: {
-    DISABLED: boolean;
-    RUN_DATA_MIGRATIONS: boolean;
+  seeds: {
+    directory: string;
+    loadExtensions: string[];
   };
+}
 
-  // inspect.ts
-  INSPECT?: {
-    DEPTH?: number;
-    SHOW_HIDDEN?: boolean;
-    COLOR?: boolean;
+interface ServiceConfig {
+  PORT: number;
+  HOST: string;
+  PARTICIPANT_ID: string;
+  DATABASE?: DatabaseConfig;
+  SHOULD_USE_IN_MEMORY_DB: string;
+  INSPECT: {
+    DEPTH: number;
+    SHOW_HIDDEN: boolean;
+    COLOR: boolean;
   };
 }
 
-const RC = parse(rc('AS', Config)) as ServiceConfig
+const ConvictConfig = Convict<ServiceConfig>({
```

Comment:
there is a need to add ENV variable like it has been done here: https://github.com/mojaloop/thirdparty-scheme-adapter/blob/master/src/shared/config.ts#L121